pkg/cloudprovider/types.go

```diff
@@ -70,7 +70,7 @@ type CapacityConstraints struct {
 	// Overhead resources per node from system resources such a kubelet and daemonsets.
 	Overhead v1.ResourceList
 	// Architecture constrains the underlying hardware architecture.
-	Architecture *Architecture
+	Architecture Architecture
```

Comment:
Why remove here, but add to CapacityConstraints?

Reply:
We have `Linux386` defined as a const -
```
const (
	Linux386 Architecture = "linux/386"		
        ...
)
```

When I try to set it in this struct, Go doesn't let me use the address of a const. So had to change this to not be a pointer.